itest/supply_commit_test.go

```diff
@@ -132,3 +136,238 @@ func testPreCommitOutput(t *harnessTest) {
+		fetchResp, err = t.tapd.FetchSupplyCommit(
+			ctxb, &unirpc.FetchSupplyCommitRequest{
+				GroupKey: groupKeyBytes,
+				IgnoreLeafKeys: [][]byte{
+					respIgnore.LeafKey,
+				},
+			},
+		)
+		require.NoError(t.t, err)
+
+		// Once the ignore tree includes the ignored asset outpoint, we
+		// know that the supply commitment has been updated.
+		return fetchResp.IgnoreSubtreeRoot.RootNode.RootSum ==
+			int64(sendAssetAmount)
+	}, defaultWaitTimeout, time.Second)
+
+	// Verify that the supply commitment tree commits to the ignore subtree.
+	supplyCommitRootHash := fn.ToArray[[32]byte](
+		fetchResp.SupplyCommitmentRoot.RootHash,
+	)
+
+	// Formulate the ignore leaf node as it should appear in the supply
+	// tree.
+	supplyTreeIgnoreLeafNode := mssmt.NewLeafNode(
+		fetchResp.IgnoreSubtreeRoot.RootNode.RootHash,
+		uint64(fetchResp.IgnoreSubtreeRoot.RootNode.RootSum),
+	)
+
+	ignoreRootLeafKey := fn.ToArray[[32]byte](
+		fetchResp.IgnoreSubtreeRoot.SupplyTreeLeafKey,
+	)
+
+	AssertInclusionProof(
+		t, supplyCommitRootHash,
+		fetchResp.IgnoreSubtreeRoot.SupplyTreeInclusionProof,
+		ignoreRootLeafKey, supplyTreeIgnoreLeafNode,
+	)
+
+	// Unmarshal ignore tree leaf inclusion proof to verify that the
+	// ignored asset outpoint is included in the ignore tree.
+	require.Len(t.t, fetchResp.IgnoreLeafInclusionProofs, 1)
+	inclusionProofBytes := fetchResp.IgnoreLeafInclusionProofs[0]
+
+	// Verify that the ignore tree root can be computed from the ignore leaf
+	// inclusion proof.
+	expectedIgnoreSubtreeRootHash := fn.ToArray[[32]byte](
+		fetchResp.IgnoreSubtreeRoot.RootNode.RootHash,
+	)
+
+	ignoreLeafKey := fn.ToArray[[32]byte](respIgnore.LeafKey)
+	ignoreLeaf := unmarshalMerkleSumNode(respIgnore.Leaf)
+
+	AssertInclusionProof(
+		t, expectedIgnoreSubtreeRootHash, inclusionProofBytes,
+		ignoreLeafKey, ignoreLeaf,
+	)
+
+	// Verify that the mined supply commitment transaction commits to the
+	// supply commitment tree.
+	require.Len(t.t, minedBlocks, 1)
+
+	block := minedBlocks[0]
+	// TODO(ffranr): Why are there two transactions in this block?
+	require.Len(t.t, block.Transactions, 2)
+
+	internalKey, err := btcec.ParsePubKey(fetchResp.AnchorTxOutInternalKey)
+	require.NoError(t.t, err)
+
+	expectedTxOut, _, err := supplycommit.RootCommitTxOut(
+		internalKey, nil, supplyCommitRootHash,
+	)
+	require.NoError(t.t, err)
+
+	foundCommitTxOut := false
+	for idx := range block.Transactions {
+		tx := block.Transactions[idx]
+
+		for idxOut := range tx.TxOut {
+			txOut := tx.TxOut[idxOut]
+
+			pkScriptMatch := bytes.Equal(
+				txOut.PkScript, expectedTxOut.PkScript,
+			)
+			if txOut.Value == expectedTxOut.Value && pkScriptMatch {
+				// Ensure that the target tx out is only present
+				// once.
+				if foundCommitTxOut {
+					t.Fatalf("found multiple supply " +
+						"commitment tx outputs in " +
+						"block")
+				}
+
+				foundCommitTxOut = true
+			}
+		}
+	}
+
+	require.True(t.t, foundCommitTxOut)
```

Comment:
We can't assert here that we wouldn't accept the ignored asset again because that part isn't hooked up fully in `tapd` yet, right?